application/backend/src/main/resources/apps/typerefinery/components/graphs/barchart/clientlibs/functions.js

```diff
@@ -0,0 +1,204 @@
+window.Typerefinery = window.Typerefinery || {};
+window.Typerefinery.Components = Typerefinery.Components || {};
+window.Typerefinery.Components.Graphs = Typerefinery.Components.Graphs || {};
+window.Typerefinery.Components.Graphs.BarChart = Typerefinery.Components.Graphs.BarChart || {};
+window.Typerefinery.Components.Graphs.Items = Typerefinery.Components.Graphs.Items || {};
+window.Typerefinery.Theme = Typerefinery.Theme || {};
+
+
+const DEFAULT_BAR_CHART_DATA = {
+  chartData: [75000, 75000, 75000, 15000, 14000, 12000, 11000, 11500, 11000],
+  labelName: "Typerefinery Bar Chart",
+  barbackgroundcolor: [
+    "#FFB94E",
+    "#FFB94E",
+    "#FFB94E",
+    "#FFB94E",
+    "#FFB94E",
+    "#ED7117",
+    "#FFB94E",
+    "#228B22",
+    "#228B22",
+  ],
+  labels: [
+    "Trazen:Win32/Qakbot",
+    "Ranson:Win32/Egre..",
+    "Quakbot",
+    "Android",
+    "Emotet",
+    "XAgentOSX",
+    "Pteranodon",
+    "PsExec",
+    "Minikatz",
+  ],
+    dataSetBorderColor:"#0099DE",
+    canvasBackgroundColor:"#99ffff"
+}
+
+    ; (function (ns, componentNs, themeNs, graphItemsNs, document, window) {
+        "use strict";
+
+        ns.updateComponentHTML = (data, $component) => {
+            if (!$component) {
+                console.log('[barchart/clientlibs/functions.js] component does not exist')
+                return;
+            }
+            console.log("bardata",data)
+            let componentConfig = componentNs.getComponentConfig($component);
+            componentConfig = {
+                ...componentConfig,
+                ...DEFAULT_BAR_CHART_DATA
```

Comment:
Nothing inside the namespaces should access the variables outside the scope.

Reply:
done